### ISSUE TITLE
IPSec deployment failing on nonexistent/nonauthorized CPE

### ISSUE THREAD

**Issue by VAdamec** [oracle/terraform-provider-baremetal#107]
## Description of the bug:

Testing deployment of VCN with DRG and IPSec, failing on IPSec deployment. Manually via UI I can set it up without any problem, so it's just some OSIC mismatch I believe.

Can you please advice ?

```
* baremetal_core_cpe.cpetest: Code: NotAuthorizedOrNotFound; OPC Request ID: /xxxxxxx/xxxxxxx; Message: Either Customer Premise Equipment(CPE) with ID ocid1.compartment.oc1..xxxxxxx does not exist or you are not authorized to access it.
```

## A self-contained terraform file that reproduces the issue:

```
variable "tenancy_ocid" {}
variable "user_ocid" {}
variable "fingerprint" {}
variable "private_key_path" {}
variable "compartment_ocid" {}

provider "baremetal" {
  tenancy_ocid = "${var.tenancy_ocid}"
  user_ocid = "${var.user_ocid}"
  fingerprint = "${var.fingerprint}"
  private_key_path = "${var.private_key_path}"
}

variable "VPC-CIDR" {
  default = "192.168.104.0/24"
}

variable "IPSEC-CPE" {
  default = "123.123.123.123"
}

resource "baremetal_core_virtual_network" "CompleteVCN" {
  compartment_id = "${var.compartment_ocid}"
  cidr_block = "${var.VPC-CIDR}"
  display_name = "CompleteVCN"
}

resource "baremetal_core_drg" "drgtest" {
    compartment_id = "${var.compartment_ocid}"
    display_name = "DrgTest"
}

resource "baremetal_core_cpe" "cpetest" {
    compartment_id = "${var.compartment_ocid}"
    display_name = "cpetest"
    ip_address = "${var.IPSEC-CPE}"
}

resource "baremetal_core_drg_attachment" "DrgAttached" {
    compartment_id = "${var.compartment_ocid}"
    display_name = "DrgTest"
    drg_id = "${baremetal_core_drg.drgtest.id}"
    vcn_id = "${baremetal_core_virtual_network.CompleteVCN.id}"
}

resource "baremetal_core_ipsec" "ipsec" {
    compartment_id = "${var.compartment_ocid}"
    cpe_id = "${baremetal_core_cpe.cpetest.id}"
    drg_id = "${baremetal_core_drg.drgtest.id}"
    display_name = "IPsecTest"
    static_routes = ["10.0.0.0/24"]
}
```

**Comment by VAdamec** [comment]
Hm, so explicit deps need to be added:

```
resource "baremetal_core_ipsec" "ipsec" {
    compartment_id = "${var.compartment_ocid}"
    cpe_id = "${baremetal_core_cpe.cpetest.id}"
    drg_id = "${baremetal_core_drg.drgtest.id}"
    display_name = "IPsecTest"
    static_routes = ["10.0.0.0/24"]
    depends_on = ["baremetal_core_cpe.cpetest","baremetal_core_drg.drgtest"]
}
```

**Comment by VAdamec** [comment]
But such `workaround` means complication during destroying resources. Any advice ?

**Comment by craigcarl-oracle** [comment]
@VAdamec the provider should be handling these dependencies, this looks like a bug. I don't think your workaround will complicate destroys. TF does a pretty good job of understanding dependencies, it should handle your workaround well.

**Comment by VAdamec** [comment]
Well that's the result, without deps it's impossible to deploy, with deps destroying need to be done multiple time (IPSec and DRG order).

**Comment by VAdamec** [comment]
First attempt:

```
* baremetal_core_cpe.cpetest (destroy): 1 error(s) occurred:
* baremetal_core_cpe.cpetest: Code: IncorrectState; OPC Request ID: /xxxxxxx/xxxxxxx; Message: ocid1.cpe.oc1.phx.xxxxxxx is associated with IPsec Connection that is in use
```

Second attempt:

```
baremetal_core_cpe.cpetest: Destroying... (ID: ocid1.cpe.oc1.phx.xxxxxxx)
baremetal_core_cpe.cpetest: Destruction complete
```

**Comment by codycushing** [comment]
Fixed in v1.0.12
